File: app/schemas/api_v1/tasks.py
"""Task API schemas."""

from __future__ import annotations

from typing import Any

from pydantic import Field

from app.schemas.api_v1.base import StrictRequest


class TaskCreateRequest(StrictRequest):
    group_id: str
    description: str = Field(min_length=1)
    worker_id: str | None = None
    reviewer_ids: list[str] = Field(default_factory=list)
    max_rounds: int = Field(default=5, ge=1)
    context: list[str] = Field(default_factory=list)
    guardrails: list[dict[str, Any]] = Field(default_factory=list)
    human_review_required: bool = False
    output_schema: dict[str, Any] = Field(default_factory=dict)

File: frontend-react/src/types/tasks.ts
import type { JsonObject, OkResult } from './common';
import type { TaskDetail } from './api';

export interface TaskGuardrail {
  name: string;
  description: string;
}

export interface CreateTaskInput {
  group_id?: string;
  groupId?: string;
  description: string;
  worker_id?: string;
  reviewer_ids?: string[];
  max_rounds?: number;
  context?: string[];
  guardrails?: TaskGuardrail[];
  human_review_required?: boolean;
  output_schema?: JsonObject;
  provider?: string;
  model_id?: string;
  api_key?: string;
}

export interface CreatedTask extends Omit<TaskDetail, 'current_round'> {
  task_id: string;
}

export interface TaskActionResult extends OkResult {
  task_id: string;
  status: string;
}

export type TaskRunInputs = JsonObject;
